src/aof.c

```diff
@@ -826,8 +827,12 @@ int openNewIncrAofForAppend(void) {
     /* If reaches here, we can safely modify the `server.aof_manifest`
      * and `server.aof_fd`. */
 
-    /* Close old aof_fd if needed. */
-    if (server.aof_fd != -1) bioCreateCloseJob(server.aof_fd);
+    /* fsync and close old aof_fd if needed. */
```

Comment:
maybe we should explain some of our concerns here.
e.g. that in fsync every-sec it's ok to delay the fsync as long as we grantee it happens, and in fsync always the file is already synced at this point so fsync doesn't matter.